exla/lib/exla/plugin.ex

```diff
@@ -0,0 +1,29 @@
+defmodule EXLA.Plugin do
+  @moduledoc """
+  Plugin system for registering custom calls.
+  """
+
+  def register(library_path) do
+    unless File.exists?(library_path) do
+      raise ArgumentError, "#{library_path} does not exist"
+    end
+
+    case :persistent_term.get({__MODULE__, library_path}, nil) do
+      nil ->
+        ref =
+          library_path
+          |> EXLA.NIF.load_custom_call_plugin_library()
+          |> unwrap!()
+
+        # we need to keep the ref from getting garbage collected so
+        # we can use the symbols within it at anytime 
+        :persistent_term.put({__MODULE__, library_path}, ref)
+
+      _ref ->
+        :ok
+    end
```

Comment:
Should we use a process instead such that, if someone does `Application.stop(:exla)` the process is shutdown as well as all plugins? 🤔

Reply:
Regarding how to store things, I would rather do an ETS than a process for storing the custom call registry if persistent term is not what we want, to keep close to the same read concurrency.

Another possible alternative would be a GenServer that manages the persistent term on terminate, it would clean up the persitent term state.

Reply:
In this manager alternative, we'd have non-process functions that read first and write if needed to the persistent term, and the only purpose for the GenServer would be to ensure cleanup upon termination.

PS: Upon writing this I went reading and found https://hexdocs.pm/elixir/1.12/Application.html#c:prep_stop/1 which would serve this purpose nicely.

Reply:
prep_stop also works but you would need to iterate all persistent term to find the keys relevant to us. ETS would be better.

Reply:
If there's something like EXLA.CustomCalls.cleanup/0 we could call, then that function will already know about all of the keys that should be cleaned up.

I see no issue with using ETS however, as the speed difference here only matters at defn compile time and not runtime